lib/vehicle.rb

```diff
@@ -1,18 +1,18 @@
 require 'date'
 
 class Vehicle
-  attr_reader :vin,
-              :year,
-              :make,
-              :model,
-              :engine
+  attr_reader :vin, :year, :make, :model, :engine, :registration_date, :county
+  attr_accessor :registration_date, :plate_type, :vin, :year, :make, :model, :engine, :county
```

Comment:
Similar to what I mentioned earlier, having the same instance variables in the `attr_reader` and `attr_accessor` is redundant. If you intent to have reading and writing of an instance variable then only an `attr_writer` is necessary.

I am also unsure of what we need to create the ability to write to the instance variables from outside of this class, since the only variables then get written to from outside of the class are `registration_date` and `plate_type`. These other variables in your `attr_accessor` can probably be removed.